dot-net-sdk/EppoClient.cs

```diff
@@ -99,7 +105,7 @@ public class EppoClient
                 .LogAssignment(new AssignmentLogData(
                     flagKey,
                     rule.allocationKey,
-                    assignedVariation.typedValue.StringValue(),
+                    assignedVariation.StringValue(),
```

Comment:
check if `assignmentVariation` is not null before calling the logger

Reply:
Since `HasEppoValue` can still carry a null string, we coalesce to the string `null` to avoid compiler warning and having to change the IAssignmentLogger interface.